Corona-Warn-App/src/main/java/de/rki/coronawarnapp/environment/download/DownloadCDNModule.kt

```diff
@@ -0,0 +1,50 @@
+package de.rki.coronawarnapp.environment.download
+
+import dagger.Module
+import dagger.Provides
+import dagger.Reusable
+import de.rki.coronawarnapp.BuildConfig
+import de.rki.coronawarnapp.diagnosiskeys.server.LocationCode
+import de.rki.coronawarnapp.http.HttpClientDefault
+import okhttp3.ConnectionSpec
+import okhttp3.OkHttpClient
+import okhttp3.TlsVersion
+import javax.inject.Singleton
+
+@Module
+class DownloadCDNModule {
+
+    @Reusable
+    @DownloadCDNHttpClient
+    @Provides
+    fun cdnHttpClient(@HttpClientDefault defaultHttpClient: OkHttpClient): OkHttpClient =
+        defaultHttpClient.newBuilder().connectionSpecs(DOWNLOAD_CDN_CONNECTION_SPECS).build()
+
+    @Singleton
+    @DownloadCDNServerUrl
+    @Provides
+    fun provideDownloadServerUrl(): String {
+        val url = BuildConfig.DOWNLOAD_CDN_URL
+        if (!url.startsWith("https://")) throw IllegalStateException("Innvalid: $url")
```

Comment:
I think maybe we should either do a proper URL check or none at all, instead of a simple https check in the string

Reply:
What would we check though? Btw, the current check is the same we had before, I just moved the code a bit.

Reply:
Okay I think this is a super unimportant todo, for now should not matter.